src/game/overlay.c

```diff
@@ -78,12 +80,90 @@ static RGB888 m_ColorBarMap[][COLOR_STEPS] = {
       { 120, 60, 70 } },
 };
 
+typedef struct BAR_INFO {
+    int32_t type;
+    int32_t value;
+    int32_t max_value;
+    int32_t show;
+    int32_t show_mode;
+    int32_t blink;
+    int32_t timer;
+    int32_t color;
+    int32_t bar_location;
+} BAR_INFO;
+
+static BAR_INFO m_HealthBar = { 0 };
+static void Overlay_SetupHealthBar()
+{
+    m_HealthBar.type = BT_LARA_HEALTH;
+    m_HealthBar.value = LARA_HITPOINTS;
+    m_HealthBar.max_value = LARA_HITPOINTS;
+    m_HealthBar.show_mode = g_Config.healthbar_showing_mode;
+    m_HealthBar.show = 0;
+    m_HealthBar.blink = 0;
+    m_HealthBar.timer = 40;
+    m_HealthBar.color = g_Config.healthbar_color;
+    m_HealthBar.bar_location = g_Config.healthbar_location;
+}
+
+static BAR_INFO m_AirBar = { 0 };
+static void Overlay_SetupAirBar()
+{
+    m_AirBar.type = BT_LARA_AIR;
+    m_AirBar.value = LARA_AIR;
+    m_AirBar.max_value = LARA_AIR;
+    m_AirBar.show_mode = g_Config.airbar_showing_mode;
+    m_AirBar.show = 0;
+    m_AirBar.blink = 0;
+    m_AirBar.timer = 0;
+    m_AirBar.color = g_Config.airbar_color;
+    m_AirBar.bar_location = g_Config.airbar_location;
+}
+
+static BAR_INFO m_EnemyBar = { 0 };
+static void Overlay_SetupEnemyBar()
+{
+    m_EnemyBar.type = BT_ENEMY_HEALTH;
+    m_EnemyBar.value = 0;
+    m_EnemyBar.max_value = 0;
+    m_EnemyBar.show_mode = g_Config.healthbar_showing_mode;
+    m_EnemyBar.show = g_Config.enable_enemy_healthbar;
+    m_EnemyBar.blink = 0;
+    m_EnemyBar.timer = 0;
+    m_EnemyBar.color = g_Config.enemy_healthbar_color;
+    m_EnemyBar.bar_location = g_Config.enemy_healthbar_location;
+}
+
 static void Overlay_GetBarLocation(
     int8_t bar_location, int32_t width, int32_t height, int32_t *x, int32_t *y);
-static void Overlay_DrawBar(int32_t value, int32_t value_max, int32_t bar_type);
+static void Overlay_DrawBar(BAR_INFO *draw_bar);
 static void Overlay_OnAmmoTextRemoval(const TEXTSTRING *textstring);
 static void Overlay_OnFPSTextRemoval(const TEXTSTRING *textstring);
 
+static void Overlay_GetBlink(BAR_INFO *blink_bar)
```

Comment:
Should I make the function bars (like `blink_bar`) all the same name since they are passed as pointers? I never know what people prefer to name as far as different function but same variables go.

Reply:
Let's just name it `bar` or `bar_info` across all functions :)